src/main/java/edu/hm/hafner/analysis/parser/PnpmAuditParser.java

```diff
@@ -0,0 +1,124 @@
+    private static final String PNPM_VULNERABILITY_SEVERITY_HIGH = "high";
+    private static final String PNPM_VULNERABILITY_SEVERITY_CRITICAL = "critical";
+
+    private static final long serialVersionUID = 4140706319863200922L;
+
+    @Override
+    protected void parseJsonObject(final Report report, final JSONObject jsonReport, final IssueBuilder issueBuilder) {
+        JSONObject results = jsonReport.optJSONObject("advisories");
+        if (results != null) {
+            parseResults(report, results, issueBuilder);
+        }
+    }
+
+    private void parseResults(final Report report, final JSONObject jsonReport, final IssueBuilder issueBuilder) {
+        for (String key : jsonReport.keySet()) {
+            JSONObject vulnerability = (JSONObject) jsonReport.get(key);
+
+            if (!vulnerability.isEmpty()) {
+                report.add(convertToIssue(vulnerability, issueBuilder));
+            }
+        }
+    }
+
+    private Issue convertToIssue(final JSONObject vulnerability, final IssueBuilder issueBuilder) {
+        return issueBuilder.setModuleName(vulnerability.optString("module_name", VALUE_NOT_SET))
+                .setCategory(formatCategory(vulnerability))
+                .setSeverity(mapSeverity(vulnerability.optString("severity", "UNKNOWN")))
+                .setType(mapType(vulnerability))
+                .setMessage(vulnerability.optString("overview", "UNKNOWN"))
+                .setDescription(formatDescription(vulnerability))
+                .buildAndClean();
+    }
+
+    private String mapType(final JSONObject vulnerability) {
+        final String cve = (String) vulnerability.optJSONArray("cves").opt(0);
+
+        return cve != null ? cve : VALUE_NOT_SET;
+    }
+
+    private String formatCategory(final JSONObject vulnerability) {
+        String moduleName = vulnerability.optString("module_name");
+        String title = vulnerability.optString("title");
+
+        if (moduleName != null && title != null) {
+            return title.replace(moduleName, "").replace(" in ", "");
+        }
+
+        return VALUE_NOT_SET;
+    }
+
+    @SuppressFBWarnings("IMPROPER_UNICODE")
+    private Severity mapSeverity(final String string) {
+        if (PNPM_VULNERABILITY_SEVERITY_INFO.equalsIgnoreCase(string)) {
+            return Severity.WARNING_LOW;
+        }
+        else if (PNPM_VULNERABILITY_SEVERITY_LOW.equalsIgnoreCase(string)) {
+            return Severity.WARNING_LOW;
+        }
+        else if (PNPM_VULNERABILITY_SEVERITY_MODERATE.equalsIgnoreCase(string)) {
+            return Severity.WARNING_NORMAL;
+        }
+        else if (PNPM_VULNERABILITY_SEVERITY_HIGH.equalsIgnoreCase(string)) {
+            return Severity.WARNING_HIGH;
+        }
+        else if (PNPM_VULNERABILITY_SEVERITY_CRITICAL.equalsIgnoreCase(string)) {
+            return Severity.ERROR;
+        }
+        else {
+            return Severity.ERROR;
```

Comment:
I don't know the severities of PnpmAudit, so my intention might be wrong: typically the default is `NORMAL`.

Reply:
I agree with you. I oriented on the existing implementation of Trivy, which is kind of similar as it also handles vulnerabilities. 

However, actually every finding should have a severity. If not it (for whatever reason), it now defaults to `normal` as suggested.